src/Symfony/Component/HttpClient/NativeHttpClient.php

```diff
@@ -258,7 +258,7 @@ private static function getProxy(?string $proxy, array $url): ?array
     {
         if (null === $proxy) {
             // Ignore HTTP_PROXY except on the CLI to work around httpoxy set of vulnerabilities
-            $proxy = $_SERVER['http_proxy'] ?? ('cli' === \PHP_SAPI ? $_SERVER['HTTP_PROXY'] ?? null : null) ?? $_SERVER['all_proxy'] ?? $_SERVER['ALL_PROXY'] ?? null;
+            $proxy = $_SERVER['http_proxy'] ?? (\in_array(\PHP_SAPI, ['cli', 'phpdbg', 'embed'])? $_SERVER['HTTP_PROXY'] ?? null : null) ?? $_SERVER['all_proxy'] ?? $_SERVER['ALL_PROXY'] ?? null;
```

Comment:
what about cli-server which is used when using phps builtin server?

Reply:
I'm not sure if we should. @kelunik what are your thoughts on it?

Reply:
That's like adding apache to the list, no? I guess the CLI server also sets these entries in case it receives an HTTP request with a `proxy` header.

Reply:
adding cli-server would expose to https://httpoxy.org/ I guess too